comp/netflow/server/component.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
It looks like we have now NetFlow code spread around 2 places:
- comp/netflow
- pkg/netflow

I think it might be best to have the code in one place if possible.

I'm thinking about two ways:

1/ Minimise as much as possible NetFlow code in `comp/netflow` by only having minimum code needed (similar to previously having only minimal code in `cmd/agent/subcommands/run/command.go`.

I think right now, we still have a lot of netflow logic in `comp/netflow`, if we go to this route 1/, I'm wondering if we can reduce it to the minimum. For example, by keeping NetFlow `Server` structure in `pkg/netflow`.

2/ Move all netflow files from `pkg/netflow` to `comp/netflow` similar to DogStatsD migration of files here: https://github.com/DataDog/datadog-agent/pull/16126


WDYT?

Reply:
Ok, I have moved all netflow code into /comp. I also did some light refactoring and added a bunch of comments to make the precommit hooks stop complaining; nothing should change the behavior.